src/libraries/System.Private.Xml/src/System/Xml/Schema/XmlValueConverter.cs

```diff
@@ -208,40 +209,40 @@ internal abstract class XmlValueConverter
         public abstract string ToString(DateTime value);
         public abstract string ToString(DateTimeOffset value);
         public abstract string ToString(object value);
-        public abstract string ToString(object value, IXmlNamespaceResolver nsResolver);
+        public abstract string ToString(object value, IXmlNamespaceResolver? nsResolver);
 
         public abstract object ChangeType(bool value, Type destinationType);
         public abstract object ChangeType(int value, Type destinationType);
         public abstract object ChangeType(long value, Type destinationType);
         public abstract object ChangeType(decimal value, Type destinationType);
         public abstract object ChangeType(double value, Type destinationType);
         public abstract object ChangeType(DateTime value, Type destinationType);
-        public abstract object ChangeType(string value, Type destinationType, IXmlNamespaceResolver nsResolver);
+        public abstract object ChangeType(string value, Type destinationType, IXmlNamespaceResolver? nsResolver);
         public abstract object ChangeType(object value, Type destinationType);
-        public abstract object ChangeType(object value, Type destinationType, IXmlNamespaceResolver nsResolver);
+        public abstract object ChangeType(object value, Type destinationType, IXmlNamespaceResolver? nsResolver);
     }
 
     internal abstract class XmlBaseConverter : XmlValueConverter
     {
-        private readonly XmlSchemaType _schemaType;
+        private readonly XmlSchemaType? _schemaType;
         private readonly XmlTypeCode _typeCode;
-        private readonly Type _clrTypeDefault;
+        private readonly Type? _clrTypeDefault;
 
         protected XmlBaseConverter(XmlSchemaType schemaType)
         {
             // XmlValueConverter is defined only on types with simple content
-            XmlSchemaDatatype datatype = schemaType.Datatype;
+            XmlSchemaDatatype? datatype = schemaType.Datatype;
             Debug.Assert(schemaType != null && datatype != null, "schemaType or schemaType.Datatype may not be null");
 
             while (schemaType != null && !(schemaType is XmlSchemaSimpleType))
             {
-                schemaType = schemaType.BaseXmlSchemaType;
+                schemaType = schemaType.BaseXmlSchemaType!;
```

Comment:
NIT: not needed `!`

Reply:
yes and no, schemaType happens to be used as an input argument and is asserted to be not null but then later is being assigned nullable, I'd need to local to express that which felt like unnecessary refactoring considering size of this PR